src/ngraph/frontend/onnx_import/op/hardmax.cpp

```diff
@@ -35,16 +36,25 @@ namespace ngraph
                 {
                     const auto input = node.get_ng_inputs().at(0);
                     const auto& input_shape = input->get_output_partial_shape(0);
-                    const auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
 
-                    const auto normalized_axis =
-                        ngraph::normalize_axis(node.get_description(), axis, input_shape.rank());
+                    auto axis = node.get_attribute_value<std::int64_t>("axis", 1);
+                    if (input_shape.rank().is_static())
+                    {
+                        axis = ngraph::normalize_axis(
+                            node.get_description(), axis, input_shape.rank());
+                    }
 
                     // reshape to 2D - "batch size" x "input feature dimensions" (NxD)
-                    const auto coerced_tensor =
-                        ngraph::builder::opset1::flatten(input, normalized_axis);
-                    const auto& coerced_shape = coerced_tensor->get_shape();
-                    const auto row_size = static_cast<int64_t>(coerced_shape.at(1));
+                    const auto coerced_tensor = ngraph::builder::opset1::flatten(input, axis);
+
+                    const auto coerced_tensor_shape =
+                        std::make_shared<default_opset::ShapeOf>(coerced_tensor);
+                    std::shared_ptr<ngraph::Node> row_size =
+                        std::make_shared<default_opset::Gather>(
+                            coerced_tensor_shape,
+                            ngraph::opset1::Constant::create(element::i64, {1}, {1}),
+                            ngraph::op::Constant::create(element::i64, {}, {0}));
```

Comment:
Why the differences in namespaces? We should use `default_opset`

Reply:
right ;)